docs/Reference/CLI/CLI-Subcommands.md

```diff
@@ -84,6 +84,247 @@ When
+The default is `false`.
+
+#### `aws-secrets-prefixes-filter`
+
+=== "Syntax"
+
+    ```bash
+    --aws-secrets-prefixes-filter=<STRING>[,<STRING>,...]
+    ```
+
+=== "Example"
+
+    ```bash
+    --aws-secrets-prefixes-filter=prefix1,prefix2
+    ```
+
+=== "Environment variable"
+
+    ```bash
+    WEB3SIGNER_ETH2_AWS_SECRETS_PREFIXES_FILTER=prefix1,prefix2
+    ```
+
+=== "Configuration file"
+
+    ```bash
+    eth2.aws-secrets-prefixes-filter: ["prefix1","prefix2"]
+    ```
+
+Optional comma-separated list of secret name prefixes filter to apply while fetching secrets from
+AWS Secrets Manager.
+
+#### `aws-secrets-region`
+
+=== "Syntax"
+
+    ```bash
+    --aws-secrets-region=<STRING>
+    ```
+
+=== "Example"
+
+    ```bash
+    --aws-secrets-region=us-east-2
+    ```
+
+=== "Environment variable"
+
+    ```bash
+    WEB3SIGNER_ETH2_AWS_SECRETS_REGION=us-east-2
+    ```
+
+=== "Configuration file"
+
+    ```bash
+    eth2.aws-secrets-region: "us-east-2"
+    ```
+
+AWS region where AWS Secrets Manager is available.
+
+Required when [`--aws-secrets-auth-mode`](#aws-secrets-auth-mode) is `SPECIFIED`.
+
+#### `aws-secrets-secret-access-key`
+
+=== "Syntax"
+
+    ```bash
+    --aws-secrets-secret-access-key=<STRING>
+    ```
+
+=== "Example"
+
+    ```bash
+    --aws-secrets-secret-access-key=wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY
+    ```
+
+=== "Environment variable"
+
+    ```bash
+    WEB3SIGNER_ETH2_AWS_SECRETS_SECRET_ACCESS_KEY=wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY
+    ```
+
+=== "Configuration file"
+
+    ```bash
+    eth2.aws-secrets-secret-access-key: "wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY"
+    ```
+
+AWS secret access key to authenticate AWS Secrets Manager.
+
+Required when [`--aws-secrets-auth-mode`](#aws-secrets-auth-mode) is `SPECIFIED`.
+
+#### `aws-secrets-tag-names-filter`
+
+=== "Syntax"
+
+    ```bash
+    --aws-secrets-tag-names-filter=<STRING>[,<STRING>,...]
+    ```
+
+=== "Example"
+
+    ```bash
+    --aws-secrets-tag-names-filter=tagName1,tagName2
+    ```
+
+=== "Environment variable"
+
+    ```bash
+    WEB3SIGNER_ETH2_AWS_SECRETS_TAG_NAMES_FILTER=tagName1,tagName2
+    ```
+
+=== "Configuration file"
+
+    ```bash
+    eth2.aws-secrets-tag-names-filter: ["tagName1","tagName2"]
+
+    ```
+
+Optional comma-separated list of tag names filter to apply while fetching secrets from AWS Secrets
+Manager.
+
+#### `aws-secrets-tag-values-filter`
+
+=== "Syntax"
+
+    ```bash
+    --aws-secrets-tag-values-filter=<STRING>[,<STRING>,...]
+    ```
+
+=== "Example"
+
+    ```bash
+    --aws-secrets-tag-values-filter=tagValue1,tagValue2
+    ```
+
+=== "Environment variable"
+
+    ```bash
+    WEB3SIGNER_ETH2_AWS_SECRETS_TAG_VALUES_FILTER=tagValue1,tagValue2
+    ```
+
+=== "Configuration file"
+
+    ```bash
+    eth2.aws-secrets-tag-values-filter: ["tagValue1","tagValue2"]
+    ```
+
+Optional comma-separated list of tag values filter to apply while fetching secrets from AWS Secrets
+Manager.
```

Comment:
At the moment, these options are written in alphabetical order (which makes sense). For ease of readability for end user, can we re-order them as:
~~~
--aws-secrets-enabled
--aws-secrets-auth-mode
~~~
followed by credentials options related with SPECIFIED auth mode i.e.
~~~
--aws-secrets-access-key-id
--aws-secrets-secret-access-key
--aws-secrets-region
~~~
followed by filter options. Filter options also need a note similar to:
```
Note: The following/above filters are applied as AND operation.
```